pandas/_libs/intervaltree.pxi.in

```diff
@@ -6,12 +6,20 @@ WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 
 from pandas._libs.algos import is_monotonic
 
-ctypedef fused scalar_t:
-    float64_t
-    float32_t
+ctypedef fused int_scalar_t:
     int64_t
     int32_t
+    float64_t
+    float32_t
```

Comment:
I think we can drop support for `float32` and `int32` dtypes here as there is no practical way to actually get an `IntervalTree` of these dtypes since `IntervalIndex` is backed by 2 indexes and we don't have a `Float32Index` or `Int32Index`, which should help build time.  Will save this for a followup though.

Reply:
Soounds good.

im confused by the naming here: why are the float dtypes included in int_scalar_t/uint_scalar_t?

Reply:
Yeah, I couldn't think of a good way to name those.  Idea is that `int_scalar_t`/`uint_scalar_t` are things that are comparable to int/uint, and we want to be able to compare against float in both cases to determine things like 0.5 being in the interval (0, 1).

I tried it without including floats and got "no matching type signature" errors when trying stuff like `IntervalTree[int].get_indexer([0.5])`.

Reply:
Wouldn’t comparisons to float be very inaccurate in the 2 ** 63 plus range where we get unsigned?

Reply:
Yes, comparisons will be inaccurate above `2**61`, so the issue is also present on the upper/lower ends of the int64 range.  Comparisons are still valid below this so probably something we'll have to live with, e.g. the same holds for comparisons against `UInt64Index` but the behavior is still allowed there.